src/Core/Content/Sitemap/Service/SitemapExporter.php

```diff
@@ -149,7 +149,7 @@ private function processSiteMapHandles(UrlResult $result): void
 
             foreach ($result->getUrls() as $url) {
                 $newUrl = clone $url;
-                $newUrl->setLoc(empty($newUrl->getLoc()) ? $host : $host . '/' . $newUrl->getLoc());
```

Comment:
can you please provide tests for this change to not produce regressions?

Reply:
Please also check when the `getLoc()` is empty or contains a slash at the end or not.

Reply:
> Please also check when the `getLoc()` is empty or if `host` contains a slash at the end or not.

This is precisely what my change is about, the `rtrim($host, '/') . '/'` is doing the trick. getLoc() returns a string, if it is empty then it's ok I think?

Reply:
> can you please provide tests for this change to not produce regressions?

Happy to do so and thought about - but would be a useful to regression to test? Please help!

We need to be aware, that this change "breaks" what Google might have indexed (i.e. before it indexed `https://xxx.tld/abc` now it is `https://xxx.tld/abc/`). Perhaps someone with some better knowledge of Google search index practices might be able to rate whether this could cause negative effects (I personally don't think so). To me, it just prevents Google to send someone to the "un-slashed" url.

Reply:
Regarding the test i guess we can have the following situations:
`$host`  has a trailing slash,
`$host` doesn't have a trailing slash,

`getLoc()` is empty,
`getLoc()` is not empty and has a trailing slash,
`getLoc()`is not empty has doesn't have a trailing slash

In a unit test i would like to see what is expected in all those combinations, to be able to judge the effect of your changes quickly

Reply:
Also is there a concrete issue if links are listed in the sitemap with or without a trailing slash? As in the end it should resolve to the exact same page right?

(you see no google search index expert here either)

Reply:
> Also is there a concrete issue if links are listed in the sitemap with or without a trailing slash? As in the end it should resolve to the exact same page right?

Yes, the non-trailing-slash-url breaks the resolution of relative links, used in Shopping Experiences' elements. To reproduce:
- create a sales channel domain like e.g. `https://xyz.tld/en`
- for the root category, add a relative link to the layout, like `<a href="some-slug">Link-Test</a>`
- point your browser to  `https://xyz.tld/en` and click on Link-Test - it will lead to  `https://xyz.tld/some-slug` - which is wrong because it is a different sales-channel-domain
- now  point your browser to  `https://xyz.tld/en/` and click on Link-Test - it will lead to  `https://xyz.tld/en/some-slug` - which is what is expected

See attached screen shots.

![image](https://github.com/shopware/shopware/assets/25815517/07d8c927-e22f-4faa-a3b5-161e8dda254d)

![image](https://github.com/shopware/shopware/assets/25815517/3640f58b-8922-4a86-8114-bc1e63a4e619)


The issue is relevant for our client, as he is re-using the CMS elements across different sales channels.

Hope the above is understandable? If not, please ask.

Reply:
> Regarding the test i guess we can have the following situations: `$host` has a trailing slash, `$host` doesn't have a trailing slash,
> 
> `getLoc()` is empty, `getLoc()` is not empty and has a trailing slash, `getLoc()`is not empty has doesn't have a trailing slash
> 
> In a unit test i would like to see what is expected in all those combinations, to be able to judge the effect of your changes quickly

I have spend some time to analyse how to test the code in SitemapExporter.php. My conclusion: Due to use of private functions (processSitemapHandle) and globals ($this->sitemapHandles) it is very difficult if not impossible to write reasonable tests. At least I would not know how, other than generation actual sitemaps and reading/interpreting them back. Which I would consider not reasonable effort. So if you insist on your test requirements in this case, I'd have to withdraw this PR, due to lack of time.

Reply:
Are you sure that your proposed fix is actually fixing the issue you are describing? 

Honestly I don't see how a different content in the sitemap.xml would lead to different links being used in the storefront :thinking:

Reply:
Why does `{{ path('frontend.home.page') }}` generate the URL with a trailing slash, but the sitemap lists the homepage without a trailing slash?

For Google, both URLs - w/ and w/o trailing slash - represent 2 different pages. With Shopware, they appear to be the same, but in fact they aren't - the base url is different (that's what you can see in my screenshots above).

And yes, my fix does solve the problem, as with the fix the sitemap entry for the homepage will have a trailing slash.

BTW, https://www.shopware.com/sitemap.xml/ get's it right :smile:

Reply:
Hi @vanWittlaer I could see the problem w or w/o the slash in relative link on storefront. However your fix is only for the exported sitemap.xml right? Could you provide an specific example where it's an issue when there's no slash in home url in sitemap.xml?

Reply:
Hi @vienthuong  Thx for having a look at this. Yes the fix is for the sitemap.xml only, since this is the only place where there is a code error re. this problem. All other locations, like when you click on the logo in the demo shop, are handled correctly, i.e. the url is resolved with the trailing slash.
If someone manually enters an url w/o trailing slash, then of course there is nothing you can do about it. For a specific example, see my screenshots from the demoshop above.
When the url is entered w/o trailing slash, the link url is not resolved correctly. i.e. it is based on the request's domain only.
When the url is entered with trailing slash, the link url is resolved correclty based on the current sales channel domain.
This behaviour is correct. The thing is just we shouldn't point a user to a wrong url (w/o trailing slash) via sitemal.xml and resulting Google index entry.